client/lib/wpcom-undocumented/lib/undocumented.js

```diff
@@ -2513,6 +2513,35 @@ Undocumented.prototype.getFeaturedPlugins = function( fn ) {
 };
 
 /**
- * Expose `Undocumented` module
+ * Fetch a nonce to use in the `updateSiteName` call
+ * @param {int}   siteId  The ID of the site for which to get a nonce.
+ * @returns {Promise}     A promise
  */
+Undocumented.prototype.getRequestSiteRenameNonce = function( siteId ) {
+	return this.wpcom.req.get( {
+		path: `/sites/${ siteId }/site-rename/nonce`,
```

Comment:
What's the purpose of this nonce? It feels like something unnecessary the way we handle authentication in Calypso - but maybe I'm missing something? It's just that I have not seen any other endpoint like this require a nonce to be fetched first.

Reply:
I'm not 100% on the details, but it's an attempt to avoid CSRF attacks. Details here: #22359
cc: @jblz

Reply:
Understood, but I'm curious why this endpoint seems to be the only one that needs this extra protection?